localstack/services/apigateway/exporter.py

```diff
@@ -0,0 +1,285 @@
+import abc
+import json
+from typing import Type
+
+from apispec import APISpec
+
+from localstack.aws.api.apigateway import ListOfModel
+from localstack.aws.connect import connect_to
+from localstack.utils.time import TIMESTAMP_FORMAT_TZ, timestamp
+
+# TODO:
+# - handle more extensions
+#   see the list in OpenAPIExt
+#   currently handled:
+#     - x-amazon-apigateway-integration
+#
+
+
+class _BaseOpenApiExporter(abc.ABC):
+    VERSION = None
+
+    def __init__(self):
+        self.export_formats = {"application/json": "to_dict", "application/yaml": "to_yaml"}
+
+    def _add_models(self, spec: APISpec, models: ListOfModel, base_path: str):
+        for model in models:
+            model_def = json.loads(model["schema"])
+            self._resolve_refs(model_def, base_path)
+            spec.components.schema(
+                component_id=model["name"],
+                component=model_def,
+            )
+
+    def _resolve_refs(self, schema: dict, base_path: str):
+        if "$ref" in schema:
+            schema["$ref"] = f'{base_path}/{schema["$ref"].rsplit("/", maxsplit=1)[-1]}'
+        for value in schema.values():
+            if isinstance(value, dict):
+                self._resolve_refs(value, base_path)
+
+    @staticmethod
+    def _get_integration(method_integration) -> dict:
+        fields = {
+            "type",
+            "passthroughBehavior",
+            "requestParameters",
+            "requestTemplates",
+            "httpMethod",
+            "uri",
+        }
+        integration = {k: v for k, v in method_integration.items() if k in fields}
+        integration["type"] = integration["type"].lower()
+        integration["passthroughBehavior"] = integration["passthroughBehavior"].lower()
+        if responses := method_integration.get("integrationResponses"):
+            integration["responses"] = {"default": responses.get("200")}
+        return integration
+
+    @abc.abstractmethod
+    def export(self, api_id: str, stage: str, export_format: str, with_extension: bool) -> str:
+        ...
+
+    @staticmethod
+    @abc.abstractmethod
+    def _add_paths(spec, resources, with_extension):
```

Comment:
This method definition seems slightly different than in the subclasses - should it be something like:
```suggestion
    @abc.abstractmethod
    def _add_paths(spec: APISpec, resources: dict, with_extension: bool):
```

Reply:
Good catch! Implementation has changed and I forgot to update the signature! Thanks a lot